connector/connect/client/jvm/src/main/scala/org/apache/spark/ml/classification/ProbabilisticClassifier.scala

```diff
@@ -123,4 +92,25 @@ abstract class ProbabilisticClassificationModel[
     //  then invoke the 'predictProbability' method of the remote model
     throw new NotImplementedError
   }
+
+  /**
+   *If the probability and prediction columns are set, this method returns the current model,
+   * otherwise it generates new columns for them and sets them as columns on a new copy of
+   * the current model
+   */
+  override private[classification] def findSummaryModel():
```

Comment:
I think we don't needs any private helper functions like this?

Reply:
I don't think so, it is used here:

```
  def evaluate(dataset: Dataset[_]): LogisticRegressionSummary = {
    // Handle possible missing or invalid prediction columns
    val (summaryModel, probabilityColName, predictionColName) = findSummaryModel()
    ...
```
and this part code we should run it in client side.